src/lib/common/JsonHelper.cpp

```diff
@@ -33,233 +34,242 @@
-     * representations. The process wouldn't be necessary if the DB couldn't hold such special characters, 
-     * but as long as we support NGSIv1, it is better to have the check (e.g. a newline could be 
-     * used in an attribute value using XML). Even removing NGSIv1, we have to ensure that the 
-     * input parser (rapidjson) doesn't inject not supported JSON characters in the DB (this needs to be
-     * investigated in the rapidjson documentation)
-     *
-     * JSON specification is a bit obscure about the need of escaping / (what they call 'solidus'). The
-     * picture at JSON specification (http://www.json.org/) seems suggesting so, but after a careful reading of
-     * https://tools.ietf.org/html/rfc4627#section-2.5, we can conclude it is not mandatory. Online checkers
-     * such as http://jsonlint.com confirm this. Looking in some online discussions
-     * (http://andowebsit.es/blog/noteslog.com/post/the-solidus-issue/ and
-     * https://groups.google.com/forum/#!topic/opensocial-and-gadgets-spec/FkLsC-2blbo) it seems that
-     * escaping / may have sense in some situations related with JavaScript code, which is not the case of Orion.
-     */
-    switch (char ch = *iter)
-    {
-    case '\\': ss << "\\\\"; break;
-    case '"': ss << "\\\""; break;    
-    case '\b': ss << "\\b"; break;
-    case '\f': ss << "\\f"; break;
-    case '\n': ss << "\\n"; break;
-    case '\r': ss << "\\r"; break;
-    case '\t': ss << "\\t"; break;
-    default:
-      /* Converting the rest of special chars 0-31 to \u00xx. Note that 0x80 - 0xFF are untouched as they
-       * correspond to UTF-8 multi-byte characters */
-      if (ch >= 0 && ch <= 0x1F)
-      {
-        static const char intToHex[16] =  { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' } ;
-
-        ss << "\\u00" << intToHex[(ch & 0xF0) >> 4] << intToHex[ch & 0x0F];
-      }
-      else
-      {
-        ss << ch;
-      }
-      break;
-    } //end-switch
-
-  } //end-for
-  ss << '"';
-  return ss.str();
+    indent = DEFAULT_JSON_INDENT;    
+  }
+  pwriter.SetIndent(' ', indent);
 }
 
 
+void JsonHelper::Null() {
+  if (indent == 0) {
+    writer.Null();
+  } else {
+    pwriter.Null();
+  }
+}
 
-/* ****************************************************************************
-*
-* vectorToJson -
-*/
-template <>
-std::string vectorToJson(std::vector<std::string> &list)
-{
-  switch (list.size())
-  {
-  case 0:
-    return "[]";
-
-  case 1:
-    return "[" + toJsonString(list[0]) + "]";
-
-  default:
-    std::ostringstream os;
-    os << '[';
-    os << toJsonString(list[0]);
-    for (std::vector<std::string>::size_type i = 1; i != list.size(); ++i)
-    {
-      os << ',' << toJsonString(list[i]);
-    }
-    os << ']';
-    return os.str();
+
+void JsonHelper::Bool(bool value) {
+  if (indent == 0) {
+    writer.Bool(value);
+  } else {
+    pwriter.Bool(value);
```

Comment:
Also, a doubt about writer/pwriter.
This code now seems to be prepared to either do "pretty printing" or "no whitespace".
How/where is the mode selected?

It would be great (personal opinion) to have a URI parameter like `?output=formated` or something similar to let the client select. This is something we've been thinking about for a long time ...